### ISSUE TITLE
wx backend canvas is not released after closed

### ISSUE THREAD

**Issue by yxdragon** [vispy/vispy#2065]
add **def __del__(self): print('wx backend canvas deleted')** to the vispy.app.backends._wx.py.CanvasBackend to monitor it.

```python
from vispy import scene
import wx

class Panel(wx.Panel):
	def __del__(self): print('panel deleted')

# scene.SceneCanvas = Panel # open this line to test the normal panel

app = wx.App()
frame1 = wx.Frame(None)
scene.SceneCanvas(parent=frame1)
frame1.Show()

frame2 = wx.Frame(None)
scene.SceneCanvas(parent=frame2)
frame2.Show()

app.MainLoop()
```
The canvas is not destroyed in time, (So the scene, with many visuals. that means the gpu memory could not be released until the program end) 

If some other object hold the canvas's ref? or we bind some event which need to be unbind?

**Comment by djhoese** [comment]
My guess is the SceneCanvas/Canvas is holding a reference to the parent Frame and they are not letting each other be garbage collected. Maybe the parent in the SceneCanvas should be a weakref if not already that way.

**Comment by djhoese** [comment]
May be relevant:

https://github.com/vispy/vispy/blob/c7f72cb8ba47f32606ec13dfab0147ca48b8a1cf/vispy/app/backends/_wx.py#L360-L372

**Comment by yxdragon** [comment]
It is just ok if we use vispy as a visual tool. just visualize a object, then close.
But memory leak is fatal for a data analysis tool like ImagePy. (open, close, analysis, show many image and times)

**Comment by djhoese** [comment]
Yeah, no one is saying this isn't an issue. However, I don't think any of the vispy maintainers have a lot of Wx experience so this bug probably won't be fixed by us any time soon. Also, I would say generally vispy applications create a Canvas and keep it around. They don't create and delete them continuously.

**Comment by yxdragon** [comment]
ok，I will try to fix it, some question about _vispy_close
I add log in _vispy_close(self), but it is not executed. 
need I call it myself when close the frame?

**Comment by yxdragon** [comment]
after much tracking work, I found the mater is _backend, the canvas keep a ref of itself.

```python
from vispy import scene
from vispy.scene import SceneCanvas
from vispy.app.backends._wx import CanvasBackend
import wx, weakref

# scene.SceneCanvas = Panel # open this line to test the normal panel
def close(evt):
	canvas = evt.GetEventObject().GetChildren()[0]
	canvas._vispy_canvas._backend = None
	evt.Skip()

app = wx.App()
frame1 = wx.Frame(None)
frame1.Bind(wx.EVT_CLOSE, close)
SceneCanvas(parent=frame1)
frame1.Show()

frame2 = wx.Frame(None)
frame2.Bind(wx.EVT_CLOSE, close)
SceneCanvas(parent=frame2)
frame2.Show()

app.MainLoop()
```

I guess this bug occur in qt and other ui frame too.
make _backend as weakref can fix it. But this involve many files.
we can have a talk about: **it whether or not would affect user interface**

**Comment by djhoese** [comment]
If you'd like to make a pull request with a fix, maybe we can discuss it there once I see the full solution.

**Comment by yxdragon** [comment]
nothing, but change _backend as a weakref.
It is not difficult to modify it, even we can use IDE 's global replacement.
but I am afraid of affecting the user interface.

**Comment by djhoese** [comment]
Why would changing this to a weakref change the user interface? A pull request would be appreciated no matter how small the change is.

**Comment by yxdragon** [comment]
we can use IDE's global replacement modify it easily, but it relate many files, include every ui backend (wx, qt, tk...).
